src/interpreter/ByteCode.h

```diff
@@ -3701,6 +3703,113 @@ class ArrayNewElem : public ArrayNewFrom {
+    ByteCodeStackOffset src3Offset() const { return m_src3Offset; }
+    uint8_t log2Size() const { return m_log2Size; }
+    const ArrayType* typeInfo() const { return m_typeInfo; }
+    uint32_t index() { return m_index; }
+    bool isNullable() const { return m_isNullable != 0; }
+
+#if !defined(NDEBUG)
+    void dump(size_t pos)
+    {
+        printf("ArrayFill ");
+        printf("index: %" PRIu32 " ", m_index);
+        DUMP_BYTECODE_OFFSET(src0Offset);
+        DUMP_BYTECODE_OFFSET(src1Offset);
+        DUMP_BYTECODE_OFFSET(src2Offset);
+        DUMP_BYTECODE_OFFSET(src3Offset);
+    }
+#endif
+
+private:
+    const ArrayType* m_typeInfo;
+    uint32_t m_index;
+    uint8_t m_log2Size;
+    uint8_t m_isNullable;
+    ByteCodeStackOffset m_src0Offset;
+    ByteCodeStackOffset m_src1Offset;
+    ByteCodeStackOffset m_src2Offset;
+    ByteCodeStackOffset m_src3Offset;
+};
+
+class ArrayCopy : public ByteCode {
+public:
+    ArrayCopy(const ArrayType* dstTypeInfo, uint32_t dstTypeIndex, bool dstIsNullable, const ArrayType* srcTypeInfo, uint32_t srcTypeIndex, bool srcIsNullable,
+              ByteCodeStackOffset src0Offset, ByteCodeStackOffset src1Offset, ByteCodeStackOffset src2Offset, ByteCodeStackOffset src3Offset, ByteCodeStackOffset src4Offset)
+        : ByteCode(Opcode::ArrayCopyOpcode)
+        , m_dstTypeInfo(dstTypeInfo)
+        , m_dstTypeIndex(dstTypeIndex)
+        , m_dstIsNullable(dstIsNullable ? 1 : 0)
+        , m_srcTypeInfo(srcTypeInfo)
+        , m_srcTypeIndex(srcTypeIndex)
+        , m_srcIsNullable(srcIsNullable ? 1 : 0)
+        , m_src0Offset(src0Offset)
+        , m_src1Offset(src1Offset)
+        , m_src2Offset(src2Offset)
+        , m_src3Offset(src3Offset)
+        , m_src4Offset(src4Offset)
+    {
+    }
+
+    ByteCodeStackOffset src0Offset() const { return m_src0Offset; }
+    ByteCodeStackOffset src1Offset() const { return m_src1Offset; }
+    ByteCodeStackOffset src2Offset() const { return m_src2Offset; }
+    ByteCodeStackOffset src3Offset() const { return m_src3Offset; }
+    ByteCodeStackOffset src4Offset() const { return m_src4Offset; }
+    const ArrayType* dstTypeInfo() const { return m_dstTypeInfo; }
+    const ArrayType* srcTypeInfo() const { return m_srcTypeInfo; }
+    uint32_t dstTypeIndex() { return m_dstTypeIndex; }
+    uint32_t srcTypeIndex() { return m_srcTypeIndex; }
+    bool dstIsNullable() const { return m_dstIsNullable != 0; }
+    bool srcIsNullable() const { return m_srcIsNullable != 0; }
+
+#if !defined(NDEBUG)
+    void dump(size_t pos)
+    {
+        printf("ArrayCopy ");
+        printf("dstTypeIndex: %" PRIu32 " ", m_dstTypeIndex);
+        printf("srcTypeIndex: %" PRIu32 " ", m_srcTypeIndex);
+        DUMP_BYTECODE_OFFSET(src0Offset);
+        DUMP_BYTECODE_OFFSET(src1Offset);
+        DUMP_BYTECODE_OFFSET(src2Offset);
+        DUMP_BYTECODE_OFFSET(src3Offset);
+        DUMP_BYTECODE_OFFSET(src4Offset);
+    }
+#endif
+
+private:
+    const ArrayType* m_dstTypeInfo;
```

Comment:
Sort the members as pointer / uint32 / uint8 to reduce the structure size (removing gaps). ByteCodeStackOffset can remain at the end